packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -33,10 +33,449 @@ export interface IVirtualNode extends cdk.IResource {
+     *
+     * @see https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HealthCheckPolicy.html
+     */
+    const HEALTH_CHECK_PROPERTY_THRESHOLDS: {[key in (keyof CfnVirtualNode.HealthCheckProperty)]?: [number, number]} = {
+      healthyThreshold: [2, 10],
+      intervalMillis: [5000, 300000],
+      port: [1, 65535],
+      timeoutMillis: [2000, 60000],
+      unhealthyThreshold: [2, 10],
+    };
+
+    if (hc === undefined) { return undefined; }
+
+    if (hc.protocol === Protocol.TCP && hc.path) {
+      throw new Error('The path property cannot be set with Protocol.TCP');
+    }
+
+    if (hc.protocol === Protocol.GRPC && hc.path) {
+      throw new Error('The path property cannot be set with Protocol.GRPC');
+    }
+
+    const healthCheck: CfnVirtualNode.HealthCheckProperty = {
+      healthyThreshold: hc.healthyThreshold || 2,
+      intervalMillis: (hc.interval || cdk.Duration.seconds(5)).toMilliseconds(), // min
+      path: hc.path || (hc.protocol === Protocol.HTTP ? '/' : undefined),
+      port: hc.port || pm.port,
+      protocol: hc.protocol || pm.protocol,
+      timeoutMillis: (hc.timeout || cdk.Duration.seconds(2)).toMilliseconds(),
+      unhealthyThreshold: hc.unhealthyThreshold || 2,
+    };
+
+    (Object.keys(healthCheck) as Array<keyof CfnVirtualNode.HealthCheckProperty>)
+      .filter((key) =>
+        HEALTH_CHECK_PROPERTY_THRESHOLDS[key] &&
+            typeof healthCheck[key] === 'number' &&
+            !cdk.Token.isUnresolved(healthCheck[key]),
+      ).map((key) => {
+        const [min, max] = HEALTH_CHECK_PROPERTY_THRESHOLDS[key]!;
+        const value = healthCheck[key]!;
+
+        if (value < min) {
+          throw new Error(`The value of '${key}' is below the minimum threshold (expected >=${min}, got ${value})`);
+        }
+        if (value > max) {
+          throw new Error(`The value of '${key}' is above the maximum threshold (expected <=${max}, got ${value})`);
+        }
+      });
+
+    return healthCheck;
+  }
+
+  /**
+   * Returns an HTTP Listener for a VirtualNode
+   */
+  public static httpNodeListener(props?: HttpNodeListenerProps): VirtualNodeListener {
+    return new HttpNodeListener(props);
+  }
+
+  /**
+   * Returns an HTTP2 Listener for a VirtualNode
+   */
+  public static http2NodeListener(props?: Http2NodeListenerProps): VirtualNodeListener {
+    return new Http2NodeListener(props);
+  }
+
+  /**
+   * Returns an GRPC Listener for a VirtualNode
+   */
+  public static grpcNodeListener(props?: GrpcNodeListenerProps): VirtualNodeListener {
+    return new GrpcNodeListener(props);
+  }
+
+  /**
+   * Returns an TCP Listener for a VirtualNode
+   */
+  public static tcpNodeListener(props?: TcpNodeListenerProps): VirtualNodeListener {
+    return new TcpNodeListener(props);
+  }
+
+  /**
+  * Binds the current object when adding Listener to a VirtualNode
+  */
+  public abstract bind(scope: cdk.Construct): CfnVirtualNode.ListenerProperty;
```

Comment:
Wrap this in a `VirtualNodeListenerConfig` interface with a single property of type `CfnVirtualNode.ListenerProperty`.